docs/tour/add-features.mdx

```diff
@@ -238,29 +207,47 @@ We've given our application the ability to perform atomic incrementation and sto
 Now let's use the atomic increment function to keep track of how many times we've greeted each person.
 
 ```go
-func (h HttpServer) Handle(request HttpRequest, responseWriter HttpResponseWriter) {
-  // Construct HttpResponse to send back
-  headers := http.NewFields()
-  httpResponse := http.NewOutgoingResponse(headers)
-  httpResponse.SetStatusCode(200)
-
-  body := httpResponse.Body().Unwrap()
-  bodyWrite := body.Write().Unwrap()
-  path := request.PathWithQuery().Unwrap() // [!code ++:7]
-
-  http.WasiLoggingLoggingLog(http.WasiLoggingLoggingLevelInfo(), "", fmt.Sprintf("Greeting %s", name))
-  bucket := http.WasiKeyvalue0_2_0_draft_StoreOpen("").Unwrap()
-  count := http.WasiKeyvalue0_2_0_draft_AtomicsIncrement(bucket, path, 1).Unwrap()
-
-  bodyWrite.BlockingWriteAndFlush([]uint8(fmt.Sprintf("Hello x%d, %s!\n", count, name))).Unwrap()
-  bodyWrite.BlockingWriteAndFlush([]uint8("Hello from Go!\n")).Unwrap() // [!code --]
-
-  // Send HTTP response
-  okResponse := http.Ok[HttpOutgoingResponse, HttpError](httpResponse)
-  bodyWrite.Drop()
-  http.StaticOutgoingBodyFinish(body, http.None[http.WasiHttp0_2_0_TypesTrailers]())
-  http.StaticResponseOutparamSet(responseWriter, okResponse)
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	atomics "github.com/wasmcloud/wasmcloud/examples/golang/components/http-hello-world/gen/wasi/keyvalue/atomics"  // [!code ++]
+	store "github.com/wasmcloud/wasmcloud/examples/golang/components/http-hello-world/gen/wasi/keyvalue/store"  // [!code ++]
+	"go.wasmcloud.dev/component/log/wasilog"  // [!code ++]
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
+
+func init() {
+	wasihttp.HandleFunc(handleRequest)
+}
+
+func handleRequest(w http.ResponseWriter, r *http.Request) {
+	logger := wasilog.ContextLogger("handleRequest")  // [!code ++]
+
+	name := r.URL.Query().Get("name")
+	if len(name) == 0 {
+		name = "World"
+	}
+	logger.Info(fmt.Sprintf("Greeting %s", name)) // [!code ++]
+
+	kvStore := store.Open("") // [!code ++:11]
+	if err := kvStore.Err(); err != nil {
+		w.Write([]byte("Error: " + err.String()))
+		return
+	}
+	value := atomics.Increment(*kvStore.OK(), name, 1)
+	if err := value.Err(); err != nil {
+		w.Write([]byte("Error: " + err.String()))
+		return
+	}
+	w.Write([]byte(fmt.Sprintf("Hello x%d, %s!\n", value, name)))
+	w.Write([]byte(fmt.Sprintf("Hello, %s!\n", name))) // [!code --]
 }
+
+//go:generate go run github.com/bytecodealliance/wasm-tools-go/cmd/wit-bindgen-go generate --world hello --out gen ./wit
+func main() {}
```

Comment:
I think it's also worth bringing the main up, and maybe even include a comment around why it's empty, because to someone familiar with Go this might look a little strange.